### DISCUSSION TITLE
Logic for constraint based on sampling

### DISCUSSION THREAD

**Discussion by Atulsingh92** [anyoptimization/pymoo#259] [Q&A]
Hello, 

I have a constraint for velocity, which was a crude definition.
`g6 = vt - 3.2`

Now, vt is calculated from sampling between 3 materials, mat1 mat2 and mat 3, for which, ideally the velocity constraint is different, given as follows. 
```
if mat1
   g6 = vt - 1.5
if mat2
   g6 = vt - 2.3
if mat3
  g6 = vt - 3.2

```

But as my sampling will contain all of the 3 materials, I am unsure what the constraint should be put as, in order to get the correct design space.  For my former definition of `g6=vt-3.2`, I could have a material 2, and still be over 2.3 , but not know about it, as my upper limit is 3.2
So is there a way I could do this?
Or 
Is the logic defined the way I described the problem, or there is some other way to do it?

```
if mat_i ==1:
  g6 = vt-1.5
elif mat_i ==2:
  g6 = vt-2.3
else
  g6 = vt-3.2
```

**Comment by blankjul** [comment]
If I understand the definition of your problem, you should probably add a categorical variable defining what material is used. Then, based on its value set your g6 constraint. So the last code snippet makes in fact sense to me.
